language/json-language-service/src/plugins/missing-asset-wrapper-plugin.ts

```diff
@@ -34,17 +34,27 @@ export class MissingAssetWrapperPlugin implements PlayerLanguageServicePlugin {
     languageService.hooks.onValidateEnd.tap(
       this.name,
       (diagnostics, { addFixableViolation, documentContext }) => {
-        // Just be naive here
-        // If there's an error for "expected asset" + an unexpected `id` and `type`, replace that with our own
-
         let filteredDiags = diagnostics;
 
-        const expectedAssetDiags = diagnostics.filter(
-          (d) =>
-            d.message.includes(
-              `Does not match any of the expected types for type: 'AssetWrapperOrSwitch'`
-            ) || d.message.startsWith("Expected property: asset")
-        );
+        const expectedAssetDiags = filteredDiags.filter((d) => {
+          const originalNode = documentContext.PlayerContent.getNodeFromOffset(
+            documentContext.document.offsetAt(d.range.start)
+          );
+
+          if (!originalNode) {
+            return false;
+          }
+
+          const hasAssetProperty =
+            originalNode.jsonNode.type === "object" &&
+            originalNode.jsonNode.children?.some(
+              (child) =>
+                child.type === "property" &&
+                child.children?.[0]?.value === "asset"
+            );
```

Comment:
This may potentially throw some false positives if there are errors thrown in non-asset objects (like `metaData` in our reference Action Asset) because there is no check to see what the expected type would be.